### ISSUE TITLE
show name of country in wrong place in Firefox and IE

### ISSUE THREAD

**Issue by ehsan-hakimi** [manifestinteractive/jqvmap#213]
Hi Sir/Madam,
Thanks for sharing your project. It work perfect.

I use div id="vmap" style="width: 1200px; height: 800px; " and in method call use    
pins: { 'au' : 'Australia', 'cn' : 'China', 'in' : 'India', 'nz' : 'New Zealand' },
pinMode: 'content'

It will work fine in Chrome and show name of these 4 country in right place but in Firefox, IE 11 and MS Edge, it show name of countries in wrong place. It seems it calculate value of top and left wrongly. Here is its html source code result in Chrome
div id="jqvmap1_cn_pin" for="cn" class="jqvmap_pin" style="position: absolute; left: 889.715px; top: 308.254px;"->China
and in comparison in Firefox
div id="jqvmap1_cn_pin" for="cn" class="jqvmap_pin" style="position: absolute; left: 991.939px; top: 578.767px;"->China

Please help me in this case, if I can do anything
![capture07](https://cloud.githubusercontent.com/assets/6956567/13485738/f3144e96-e15e-11e5-95ec-11be6f952158.PNG)

Best regards,
Ehsan

**Comment by health901** [comment]
It's not about browsers. It seems the pins are placed at the center of the  region path, so the center of the region may not on the land.
![pin](https://cloud.githubusercontent.com/assets/1503105/13589301/d0ef81da-e511-11e5-858c-fbd001780ec5.png)

**Comment by ehsan-hakimi** [comment]
Thanks Health901, 
You are right about US but my problem is happening in Australia, China and India which all should be inside the land (not in another country). Also US will show in sea in all web browsers but in my scenario they will be appear in right places in some browsers and in wrong places in another browser.
Regards,

**Comment by health901** [comment]
Hi,
I found that the css style "margin" of will effect this pins,  my solution is putting the map in an iframe.

to fix the US problem, I wrote a function, add offsets to the pins:
https://gist.github.com/health901/470a0314c95ed20aaa5e
preview
https://cdn.rawgit.com/health901/470a0314c95ed20aaa5e/raw/d82cb4183b96033926cd33234917bd7cab47eacf/JQVMap_PIN.html

**Comment by manifestinteractive** [comment]
@ehsan-hakimi and @health901 yes, this issues is exactly because the SVG shape is rectangular in shape in regards to centering ... meaning it does not care what the actual shape is ( like a state, country, etc ).  The browser gets the dimensions for where the center of both labels and pins should be placed based on the furthest edges of the top, right, bottom & left edge.

The only real solution is to apply custom CSS styling for the specific pin or label you have issues with.

There is an existing example for tweaking the placement of labels here:  https://github.com/manifestinteractive/jqvmap/blob/master/examples/labels.html

That can totally be usable for pins too.  I will add an example of v1.5.0 for custom styling these.

I will go ahead and close this out because its not really a bug.  This is just how SVG works.  The solution is custom CSS to move the label based on the offset.  Each browser can technically handle SVG slightly different depending on the core engine.

Basically since there is not a way to get a center of a shape, but rather the center of the bounding box for that shape, there is not a way to automate this.
